spec/preferences_spec.rb

```diff
@@ -0,0 +1,67 @@
+require "spec_helper"
+
+describe CF::Preferences do
+  before do
+    @validDomain = "TestValidDomain"
+    @invalidDomain = "TestInvalidDomain"
+    @validKey = "ValidTestKey"
+    @invalidKey = "InvalidTestKey"
+    @value = "TestValue"
+  end
+
+  after(:each) do
+    CF::Preferences.set!(@validKey, nil, @validDomain)
+    CF::Preferences.set!(@invalidKey, nil, @validDomain)
+  end
+
+  describe "self.set" do
+    context "when called with valid domain/default pair" do
+      it "writes correct value and returns true" do
+        expect(CF::Preferences.set(@validKey, @value, @validDomain)).to eq true
```

Comment:
All these tests touch/modify OS/filesystem where these are run. Even though you set the values back to nil but there will always be side-effects like these (and more). So, could you mock out and implement these unit tests?

Reply:
mocking out the calls wasn't actually testing anything so I had to do this. Since I'm using a fake domain, we don't need to worry about it

Reply:
example: https://github.com/chef/chef/blob/main/spec/unit/resource/macos_user_defaults_spec.rb

Reply:
> mocking out the calls wasn't actually testing anything so I had to do this. Since I'm using a fake domain, we don't need to worry about it
It depends on what you are mocking, there is plenty to unit test here and if there's nothing, then lose this test altogether (i.e. bad) but we certainly can't have it in this form. Doesn't matter if you use a fake domain, the tests interact with underlying OS's UserDefaults, and that's a side effect.

> example: https://github.com/chef/chef/blob/main/spec/unit/resource/macos_user_defaults_spec.rb
This is not testing a resource outcome but the utility methods `defaults_export_cmd` and `defaults_modify_cmd`, hence not interacting/mutating UserDefaults.